pkg/controllers/job/plugins/svc/svc.go

```diff
@@ -88,6 +89,34 @@ func (sp *servicePlugin) OnPodCreate(pod *v1.Pod, job *batch.Job) error {
 		pod.Spec.Subdomain = job.Name
 	}
 
+	var hostEnv []v1.EnvVar
+	var envNames []string
+
+	for _, ts := range job.Spec.Tasks {
+		// TODO(k82cn): The splitter and the prefix of env should be configurable.
+		envNames = append(envNames, fmt.Sprintf(EnvTaskHostFmt, strings.ToUpper(ts.Name)))
+		envNames = append(envNames, fmt.Sprintf(EnvHostNumFmt, strings.ToUpper(ts.Name)))
+	}
+
+	for _, name := range envNames {
+		hostEnv = append(hostEnv, v1.EnvVar{
+			Name: name,
+			ValueFrom: &v1.EnvVarSource{
```

Comment:
Downward api is used when we can not get the value directly, and is it not that efficient. Here can set the value directly.